src/main/java/com/marklogic/spark/writer/MarkLogicWriteDataSource.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.spark.writer;
+
+import com.marklogic.spark.MarkLogicTable;
+import org.apache.spark.sql.connector.catalog.Table;
+import org.apache.spark.sql.connector.catalog.TableProvider;
+import org.apache.spark.sql.connector.expressions.Transform;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.CaseInsensitiveStringMap;
+
+import java.util.Map;
+public class MarkLogicWriteDataSource implements TableProvider {
+    @Override
+    public StructType inferSchema(CaseInsensitiveStringMap options) {
+       // TODO: Need a story to read the schema passed in since this method is needed and it cannot return null.
```

Comment:
This is interesting - we of course don't require any schema in order for data to be written. And in our Spark 2 connector, it looks like  `StructType schema` was always passed in via `DefaultSource`. Ideally, we can support a way for the user not to have to provide a schema, but maybe that won't be possible.